### PR TITLE
fix: Resolve peer/direct dependency issues, Support styled-components 5

### PR DESCRIPTION
**Pull request by brandonlilly**
**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [x] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [x] Tested my solution on Mobile & Tablet.
- [x] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [x] Updated snapshots for all permutations (`npm test -- -u`).
- [x] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [x] Created TODOs for known edge cases.
- [x] Documented all of my changes (inline & doc site).
- [x] Made sure that all accessibility errors are resolved.
- [x] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.
- [x] Added name to OWNERS file for all new components
- [x] If adding a new component, add its export to the rollup config
- [x] package.json version is bumped (if necessary)

---------
**Outline your feature or bug-fix below**

Fixes some dependency issues, removes the ThemeProvider export, patches Storybook

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -54,27 +54,25 @@
...
-    "@storybook/addon-a11y": "^5.3.3",
-    "@storybook/addon-actions": "^5.3.3",
-    "@storybook/addon-info": "^5.3.3",
-    "@storybook/addon-knobs": "^5.3.3",
-    "@storybook/addon-links": "^5.3.3",
-    "@storybook/addon-options": "^5.3.3",
-    "@storybook/addon-storyshots": "^5.3.3",
-    "@storybook/addons": "^5.3.3",
-    "@storybook/react": "^5.3.3",
+    "@storybook/addon-a11y": "^5.3.12",
```
Comment (sharkparty) [inline package.json:67]: Aren't we trying to stamp this?

Comment (brandonlilly) [inline package.json:67]: hotfix `^5.3.12` resolves the `.toJson` issue

File: package.json [hunk 2]
```diff
@@ -122,16 +120,16 @@
...
     "ts-jest": "^24.0.2",
     "tslib": "^1.10.0",
     "tslint": "^5.20.1",
     "typescript": "^3.7.3",
     "yargs": "^14.0.0"
   },
   "dependencies": {
-    "@xstyled/styled-components": "^1.8.4",
-    "@xstyled/system": "^1.8.4",
+    "@xstyled/styled-components": "^1.16.0",
```
Comment (brandonlilly) [inline package.json:131]: `1.16.0` is an update to xstyled's peer dependencies for sc5

File: src/theme/index.ts [hunk 3]
```diff
@@ -1,7 +1,6 @@
 // Theme Files
 import { typography } from './typography.theme';
 import { ColorsTheme } from './colors.theme';
-import { ThemeProvider as XstyledThemeProvider } from '@xstyled/styled-components';
```
Comment (brandonlilly) [inline src/theme/index.ts:4]: RIP ThemeProvider

File: src/AutoComplete/ResultsContainer/__snapshots__/ResultsContainer.component.spec.tsx.snap [hunk 4]
```diff
@@ -137,7 +137,7 @@ exports[`Component: ResultsContainer should accept a resultTemplate 1`] = `
       onMouseOver={[Function]}
     >
       <span
-        className="sc-LzLte c2 lg anchor-typography"
+        className="sc-pzMyG c2 lg anchor-typography"
```
Comment (brandonlilly) [inline src/AutoComplete/ResultsContainer/__snapshots__/ResultsContainer.component.spec.tsx.snap:140]: I was worried about this. Not a huge deal, but certainly annoying.

File: package.json (hunk 1 above)
Comment (sharkparty) [reply package.json:67]: Oh that's awesome. Very quick turnaround.

**Review by sharkparty — APPROVED** [review]
LGTM

File: src/theme/index.ts (hunk 3 above)
Comment (sharkparty) [reply src/theme/index.ts:4]: ☠️

**Pull request merged**